src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -138,8 +138,22 @@ public WebSocketsTransport(WebSocketOptions options, IDuplexPipe application, De
         {
             try
             {
+                var endOfMessage = true;
```

Comment:
Why does this start as true? Doesn’t it allocate on the first read?

Reply:
I guess since we'll read the handshake it should be false

Reply:
Although, if not using Hubs then there isn't a guaranteed read

Reply:
So it should be false right?

Reply:
If we want to optimize for the Hub usage, yeah